### PR TITLE
[#14] Add tooling for unit, integration, and accessibility tests

### PR DESCRIPTION
**Pull request by braxex**
## Summary

- Configures tooling for all testing types (unit, integration, accessibility)
- Adds utilities for all testing types
- Adds baseline tests for all testing types

Addresses #14

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This is a documentation update

## How To Test
1. Pull changes
2. Run `yarn`
3. Run unit tests (`yarn test:cov`) and verify that all unit tests pass
4. Spin up locally
5. Run integration tests (`yarn test:cy:open`) and verify that all integration tests pass

## Checklist:
- [x] My code follows the style guidelines of this project and has no linting errors
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation, including any applicable ADRs
- [x] My changes generate no new warnings
- [x] I have added tests that fail without these changes
- [x] New and existing tests (unit, integration, accessibility) pass locally 
- [x] Documentation updated
- ~If there are security concerns they are addressed or ticketed after being discussed~

### CODE REVIEW

File: src/app/components/layout/PageHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { GovBanner, GridContainer, Header } from "@trussworks/react-uswds";
+// styles
+import styles from "./PageHeader.module.scss";
+
+export const PageHeader = () => {
+  return (
+    <>
+      <GovBanner />
+      <Header basic={true} className={styles.header}>
+        <div role="navigation">
```
[the remark this answers is not in the document]
Comment (braxex) [reply src/app/components/layout/PageHeader.tsx:11]: CodeClimate is not yet configured, so this is not valid. Configuration will follow in the next PR.

File: src/app/testing/jest/frontend/Homepage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const homepage = <Home />;
+
+describe("Test Homepage", () => {
+  test("Check that page renders", () => {
+    render(homepage);
+    expect(screen.getByText("App running successfully")).toBeVisible();
+  });
+});
+
+describe("Test Homepage accessibility", () => {
```
[the remark this answers is not in the document]
Comment (braxex) [reply src/app/testing/jest/frontend/Homepage.test.tsx:14]: CodeClimate is not yet configured, so this is not valid. Configuration will follow in the next PR.

File: src/app/testing/jest/frontend/PageHeader.test.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    const header = screen.getByRole("navigation");
+    expect(header).toBeVisible();
+  });
+
+  test("Title is visible", () => {
+    expect(screen.getByText("U.S. College Finder")).toBeVisible();
+  });
+});
+
+describe("Test Header accessibility", () => {
```
[the remark this answers is not in the document]
Comment (braxex) [reply src/app/testing/jest/frontend/PageHeader.test.tsx:22]: CodeClimate is not yet configured, so this is not valid. Configuration will follow in the next PR.

**Review by gmrabian — COMMENTED** [review]
I'm getting a 404 when I run the app. Did our routes change? I'm wondering if the `src/` directory messes with NextJS somehow 🤔 

Screenshot 2024-03-12 at 10 23 13 AM

File: src/app/page.tsx [hunk 4]
```diff
@@ -7,6 +7,16 @@ export default function Home() {
...
+        <h1>Welcome</h1>
+        <h1>Welcome</h1>
+        <h1>Welcome</h1>
+        <h1>Welcome</h1>
+        <h1>Welcome</h1>
+        <h1>Welcome</h1>
+        <h1>Welcome</h1>
+        <h1>Welcome</h1>
+        <h1>Welcome</h1>
+        <h1>Welcome</h1>
```
Comment (gmrabian) [inline src/app/page.tsx:19]: I think this causes an accessibility issue?

Screenshot 2024-03-12 at 11 53 40 AM

Comment (ntsummers1) [reply src/app/page.tsx:19]: Funnily enough, thats ok. Though yeah, we should reduce it to 1. It's mad that theres a role='navigation' on a div in the header. Thats not a unique landmark so its through an error. That being said, we haven't included navigation in the header yet so removing the role altogether is the best approach atm.

**Review by gmrabian — DISMISSED** [review]
cy and jest pass

**Review by gmrabian — APPROVED** [review]

**Pull request merged**